zha/application/platforms/cover/__init__.py

```diff
@@ -213,12 +213,30 @@ def restore_external_state_attributes(
         self,
         *,
         state: CoverState | None,
-        target_lift_position: int | None = None,
-        target_tilt_position: int | None = None,
+        **kwargs: Any,  # pylint: disable=unused-argument
     ):
-        """Restore external state attributes."""
+        """Restore external state attributes.
+
+        If the state is OPENING or CLOSING, a callback is scheduled
+        to determine the final state after the default timeout period.
+        """
+        if not self._state or state not in (CoverState.OPENING, CoverState.CLOSING):
+            return
+        if state == CoverState.CLOSING and self.is_closed:
+            return
+        if (
+            state == CoverState.OPENING
+            and self._state == CoverState.OPEN
+            and self.current_cover_position in (100, None)
+            and self.current_cover_tilt_position in (100, None)
+        ):
+            return
+
         self._state = state
-        # Target positions have been removed
+        self._loop.call_later(
+            DEFAULT_MOVEMENT_TIMEOUT,
+            functools.partial(self._determine_cover_state, refresh=True),
+        )
```

Comment:
To make sure we don't have lingering tasks and such, add `self._on_remove_callbacks.append(handle.cancel)` to make sure the handle is cleaned up in case the entity is removed before this timer runs.

Reply:
Yeah, this entity should probably override `async def on_remove` to cancel `self._lift_transition_timer` and `self._tilt_transition_timer`. Those are recreated more than once and would be a memory leak if added to `self._on_remove_callbacks` each time.

Reply:
I opted to add the transition clear methods as `_on_remove_callbacks` to avoid us needing to implement a custom `on_remove` method.